src/sync_protocol/src/agent_sync_protocol.cpp

```diff
@@ -40,25 +58,28 @@ void AgentSyncProtocol::synchronizeModule(const std::string& module, Mode mode,
             {
                 std::cerr << "EndAck failed for module: " << module << std::endl;
             }
+
             return;
         }
 
         const auto ranges = receiveReqRet();
+
         if (!ranges.empty())
         {
-            sendDataMessages(module, session, &ranges);
+            std::vector<PersistedData> rangeData = m_persistentQueue->fetchRange(module, ranges);
+            sendDataMessages(session, rangeData);
         }
     }
 }
 
-bool AgentSyncProtocol::sendStartAndWaitAck(const std::string& module, Mode mode, bool realtime, uint64_t& session)
+bool AgentSyncProtocol::sendStartAndWaitAck(const std::string& module, Mode mode, bool realtime, uint64_t& session, const std::vector<PersistedData>& data)
```

Comment:
Maybe it is better to add parametrize only the size?

Reply:
Done!